src/javascripts/modules/pillbar.js

```diff
@@ -1,53 +1,33 @@
 export default class Pillbar {
   constructor(el) {
     this.el = el
-    this.setVars()
-    this.setupListeners()
+    this.createObserver()
   }
 
-  setVars(el) {
-    /* Ticking essentially == isRequestingAnimationFrame; I'm leaving
-    it as 'ticking' to follow requestingAnimationFrame convention/docs */
-    this.ticking = false
-    this.lastScrollY = 0
-    this.el.pillbarOffset = this.el.offsetTop
-  }
-
-  setupListeners() {
-    window.addEventListener('resize', () => this.handleResize())
-    window.addEventListener('scroll', () => this.handleScroll())
-  }
+  createObserver() {
+    this.observer
 
-  handleScroll() {
-    this.lastScrollY = window.scrollY
-    if (!this.ticking) {
-      window.requestAnimationFrame(() => {
-        this.togglePillbar()
-      })
+    this.options = {
+      root: null,
+      rootMargin: "0px",
+      threshold: 0,
     }
-    this.ticking = true
-  }
 
-  handleResize() {
-    clearTimeout(this.timeout)
-    this.timeout = setTimeout(() => this.getPillbarOffset(), 200)
+    this.observer = new IntersectionObserver(this.handleIntersect, this.options)
+    this.observer.observe(document.querySelector('.title-box'))
   }
 
-  getPillbarOffset() {
-    if(!this.el.isPinned){
-      this.el.pillbarOffset = this.el.offsetTop
-    }
-  }
-
-  togglePillbar() {
-    this.ticking = false
-
-    if (this.lastScrollY > this.el.pillbarOffset) {
-      this.el.classList.add('pillbar-fixed')
-      this.el.isPinned = true
-    } else {
-      this.el.classList.remove('pillbar-fixed')
-      this.el.isPinned = false
-    }
+  handleIntersect(entries, observer) {
```

Comment:
Is `entries` the list of elements you're observing?

Reply:
It's the array of intersection objects that IntersectionObserver spits out when it is triggered. Each object has ~10 key-value pairs describing what was observed, including time, and isIntersecting boolean, etc.

I've logged entries and it must re-write with every intersection because entries.length is always 1

Reply:
Ah, I see it in the docs now. Thanks!